### DISCUSSION TITLE
Anyof in Cue

### DISCUSSION THREAD

**Discussion by mrg0lden** [cue-lang/cue#1111] [Q&A]
Hey, I'm trying to make something similar to JSON Schema's `anyof`, which basically means that there's a struct of optional fields, but at least one of them must be present. I tried to do:

```
#example: ({a: string} | {b: string} | {c: string}) & {a?: string, b?: string, c?: string}
```

but it didn't work, I get `failed to encode: ... incomplete value ...` (... is removed for readability)

**Comment by myitcv** [comment]
This is an area that we're working on. Please search for the "AnyOn" related headings in https://github.com/cue-lang/cue/issues/822

**Comment by verdverm** [comment]
@mrg0lden this may be a temporary work around

```cue
import "list"

val: this={
	a?: string
	b?: string
	c?: string
	#AnyOfABC: true & list.MinItems([for label,_ in this if list.Contains(["a", "b", "c"],label) {label}], 1)

	d?: string
	e?: string
	#AnyOfDE: true & list.MinItems([for label,_ in this if list.Contains(["d", "e"],label) {label}], 1)

	#AnyOfXYZ & { #CheckXYZ: true }

	a: "a"
	d: "d"
	z: "z"
}

#AnyOfXYZ: this={
	x?: string
	y?: string
	z?: string
	#CheckXYZ: list.MinItems([for label,_ in this if list.Contains(["x", "y", "z"],label) {label}], 1)
}
```
